src/sbefw/core/sbeConsole.C

```diff
@@ -188,27 +190,21 @@ static void uartPutChar(char c)
         if(rc != SBE_SEC_OPERATION_SUCCESSFUL)
         {
             SBE_ERROR(SBE_FUNC " LSR read error.");
-            break;
-        }
-        if(data == LSR_BAD)
-        {
+        } else if(data == LSR_BAD) {
             SBE_ERROR(SBE_FUNC " LSR_BAD data error.");
-            break;
-        }
-        if(loops >= DELAY_LOOPS)
-        {
+        } else if(loops >= DELAY_LOOPS) {
             SBE_ERROR(SBE_FUNC " FIFO timeout.");
-            break;
-        }
-
-        rc = writeReg(THR, c);
-        if(rc != SBE_SEC_OPERATION_SUCCESSFUL)
-        {
-            SBE_ERROR(SBE_FUNC " failure to write THR");
-            break;
-        }
+        } else {
+	    tx_room = 16;
+	}
+    }
 
-    } while(0);
+    rc = writeReg(THR, c);
```

Comment:
Not handling the case - when the buffer is full and if one of the errors - LSR read error, LSR_BAD or FIFO timeout.

Reply:
The current code wasn't handling that either, it would just SBE_ERROR out and skip it.

We don't do any handling of that in skiboot, so in practice it's fine.

Reply:
Stewart, at least in the original patch set, we are bailing out the writeReg basis the readReg at line  177.

At any reason why you want to continue the writeReg even for readReg failure ? 
Reason probably I see is that lower level call to lpc_rw will fail anyways while starting with the command register write to the LPC if there is an issue with readReg..

We probably can save little more cycles by not intentionally going into the writeReg if readReg fails.

Comments ?

Reply:
Stewart, Srikantha mentioned the point at line 197, where if you have data == LSR_BAD, then you will skip initialising  tx_room to 16, and we will continue to do writeReg, if succeeds the success case will do tx_room -- and initialise the count to 256...the window will increase from 16 chars to 256 chars. 

I have an opinion here to remove the readReg check altogether before every writeReg. Since inside the procedure code i.e lpc_rw, we have all the check happening. We say sucess/failure for a read/write operation only if LPC_STATUS_REG_DONE status is true/false. which is enough to identify if the current operation is success/failure.

readReg at the beginning is just adding some time buffer nothing else.

Reply:
I have pulled up this patchset into master/sbe branch, once we close on this review, I can go ahead and merge. thanks.